parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -1433,12 +1436,14 @@ public void endBlock() throws IOException {
   public void appendFile(Configuration conf, Path file) throws IOException {
     try (ParquetFileReader reader = ParquetFileReader.open(conf, file)) {
       reader.appendTo(this);
+      readers.add(reader);
```

Comment:
IIUC, ParquetFileReader implements AutoCloseable which should be closed automatically when the current scope has ended. Could you help investigate why it is not called on Windows? I have concern that this fix might hide the root cause forever.